### PR TITLE
Added a new condition, that the field is not refreshed after form-submit

### LINKED ISSUE #103: Wrong association is displayed on index
# How to reproduce:
## Prerequisites: 
- docker (or docker for mac) is installed
- dinghy http proxy (See [Readme](https://github.com/cloudstorm/cs-node-app-template/tree/cs-model#development-environment) of cs-node-app-template, Development environment chapter)
## Steps
- Check out master branch of cloudstorm
- Modify the sample.js file to use the commented out MainCtrl  (don't forget to comment out the other one)
- Check out cs-model branch of [cs-node-app-template](https://github.com/cloudstorm/cs-node-app-template/tree/cs-model) and follow the readme's instructions to get it up & running (without the frontend app) - use the cloudstorm sample app to modify the resources
- Create a few teams and a few coaches, assign the coaches to teams, so there are data in your local db
- create a new Coach and during the creation create a new Team
- The error should be visible: all coaches will have the newly created Team on the index. A refresh solves the problem, so the data on the backend is okay, but the FE datastore is messed up.

## Pictures
State before adding a new coach:
image

Adding a new coach with a newly created Team:
image

State right after adding:
image

State after a browser refresh:
image

---

This is one example of a bug, where the FE datastore is not in sync with the backend data, but there are more instances where the same problem causes similar problems. Please test the frontend app thoroughly with the provided backend where different associations are available and document all the issues you find (So create issues for the different variations of the issue after solving this one.)
A non-complete list of things to test:
- Add/delete/edit items with associations, create new items through the wizard
- Order/Filter - should be kept after adding/deleting/editing items
- Delete associated items, check the associatee

One more thing to note: There are many unnecessary calls to the backend (you can trace all backend calls in the log of cs-node-app-template), please also create issues when more calls are made to the backend that necessary.

### PR DESCRIPTION
**Pull request by kdaveed**
Resolves #103, #104

### CODE REVIEW

**Comment by kdaveed** [comment]
It works currently for the edit mode of the form.

File: src/components/cs-form/cs-form.js.coffee [hunk 1]
```diff
@@ -4,7 +4,8 @@ app = angular.module('cloudStorm.form', [])
 
 # ===== DIRECTIVE =============================================================
 
-app.directive "csForm", ['csSettings', (csSettings) ->
+app.directive "csForm", ['csSettings', 'csDescriptorService', '$rootScope',
+(csSettings, csDescriptorService, $rootScope) ->
```
Comment (bence-nagy) [inline src/components/cs-form/cs-form.js.coffee]: Remove the unnecessary injections

File: src/cs-resource-service/cs-resource-service.js.coffee [hunk 2]
```diff
@@ -35,6 +35,12 @@ app.factory 'ResourceService', ['$injector', ($injector) ->
     register_dynamic_name = (name, service_name) ->
       lookup_table[name] = service_name
 
+    setRelationships : (rels) ->
+      relationships = rels
+
+    getRelationships: () ->
+      return relationships
+
```
Comment (bence-nagy) [inline src/cs-resource-service/cs-resource-service.js.coffee]: Are those methods used?

File: src/cs-resource/cs-resource.js.coffee [hunk 3]
```diff
@@ -376,6 +392,15 @@ app.factory 'csResource', [ 'csRestApi', 'csDataStore', 'ResourceService', 'csSe
...
 
+      $removeSingleRelationship: (relationship) ->
+
+          @relationships[relationship].data = []
+
+      $getDescriptor: () ->
+
+        ResourceService.getRelationships()
+
+
```
Comment (bence-nagy) [inline src/cs-resource/cs-resource.js.coffee]: Are those used?

File: src/cs-resource/cs-resource.js.coffee [hunk 4]
```diff
@@ -327,7 +328,21 @@ app.factory 'csResource', [ 'csRestApi', 'csDataStore', 'ResourceService', 'csSe
...
+        console.log("addSelection")
+        if field.cardinality is 'one'
+          if value != undefined
+            csDescriptorService.register(this, value, "add")
+        else
+          oldValue = @relationships[field.relationship].data
+          if value.length > oldValue.length
+            csDescriptorService.register(this, value.pop(), "add")
+
+
```
Comment (bence-nagy) [inline src/cs-resource/cs-resource.js.coffee]: Is this used?

**Comment by kdaveed** [comment]
The original issue is solved. Moreover, the index table reloads after the form submission.
Finally, the handling of the new resources created on subforms, in the case of two or more association fields from the same type, works properly.

**Comment by kdaveed** [comment]
The filtering works again

**Pull request merged**